tests/integration_test.rs

```diff
@@ -298,3 +304,59 @@ async fn test_runtime_context_aware<F, Fut>(
         .send(())
         .expect("cannot send shutdown signal");
 }
+
+#[rstest]
+#[case::policy("ghcr.io/kubewarden/tests/context-aware-test-policy:latest")]
```

Comment:
I think we need to use a brand new test policy here since the context-aware one needs the Kubernetes mock scenario to run.
Maybe we could have a host capabilities test policy with to test all the other callbacks including the image manifest one.

Reply:
Agree, I suppose the changes in `ghcr.io/jvanz/policies/context-aware-test-policy:latest` will end up in `ghcr.io/kubewarden/tests/context-aware-test-policy`.

Reply:
Yes, I've started to change a policy to test this capability. But then I figured out that I can test this specific capability in isolation by spawning the callback handler alone and simulating the request sent by the evaluator. Therefore, I've cancel the policy change. Therefore, this test is not  testing the whole evaluation context it does check the OCI client integration only. Furthermore, the logic of sending request using the whole evaluation context is under test in the context aware test in the same file. Thus, I think we can create the new policy when we start to write test to test all the capabilities in the near future (considering our discussions about improving testing )

Reply:
I haven't check the content to avoid failing tests if the response from the registry (ghcr.io) change for some reason outside of out control. Furthermore, if the payload changed the serialization will fail any way. But I see the value of checking if the enum type, in other words, the content type is the expected considering the given image URI.

Reply:
Done. I've added the code to check the result type.